File: MergeStringsAlternately/mergeStringsAlternately.go
func mergeAlternately(w1 string, w2 string) string {
	res := ""
	max_len := max(len(w1), len(w2))
	w1_arr := strings.Split(w1, "")
	w2_arr := strings.Split(w2, "")

	for i := 0; i < max_len; i++ {
		if i < len(w1) {
			res += w1_arr[i]
		}
		if i < len(w2) {
			res += w2_arr[i]
		}
	}
	return res
}

File: MergeStringsAlternately/mergeStringsAlternately.py
def mergeAlternately(self, word1: str, word2: str) -> str:
        maxlen = max(len(word1), len(word2))
        result = ""
        
        for i in range(maxlen):
            result += word1[i] if i < len(word1) else ""
            result += word2[i] if i < len(word2) else ""
            
        return result
